server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSReceiveOptionsCheckUtility.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2024 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+
+package org.eclipse.lsp.cobol.implicitDialects.cics.utility;
+
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.eclipse.lsp.cobol.common.dialects.DialectProcessingContext;
+import org.eclipse.lsp.cobol.common.error.SyntaxError;
+import org.eclipse.lsp.cobol.implicitDialects.cics.CICSParser;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/** Checks CICS Receive rules for required and invalid options */
+public class CICSReceiveOptionsCheckUtility extends CICSOptionsCheckBaseUtility {
+
+  public CICSReceiveOptionsCheckUtility(
+      DialectProcessingContext context, List<SyntaxError> errors) {
+    super(context, errors);
+  }
+
+  /**
+   * Entrypoint to check CICS RECEIVE rule options
+   *
+   * @param ctx ParserRuleContext subclass containging options
+   * @param <E> A subclass of ParserRuleContext
+   */
+  public <E extends ParserRuleContext> void checkOptions(E ctx) {
+    if (ctx.getClass() == CICSParser.Cics_receive_group_oneContext.class) {
```

Comment:
Why not use a Switch statement?

Reply:
does that work with Java 8?

Reply:
very good question, I don't have a good answer except for trying to build an example using JDK8. I spent 2 days debugging java 8 issue last weak...

Reply:
Looked into it some more and the only real way to do it with a Switch is to get the strings of the class. The only other option would be to use a Map and do it that way, but as that is an uncommon way to do it might be difficult to read. 

Am just worried about the ease of maintainability and performance for future command sets. For example CICS's INQUIRE system command has 100 different versions.

Reply:
So these checks are receive specific, so different approach may be chosen to the inquire command.

Reply:
I realize that, but for the most part other developers will be using this as the basis when enforcing the grammar(s) while also allowing the flexibility that the compiler provides.